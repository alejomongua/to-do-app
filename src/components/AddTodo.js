import React from 'react'
import { connect } from 'react-redux'
import { addTodo } from '../redux/actions'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faPlusCircle } from '@fortawesome/free-solid-svg-icons'

const AddTodo = ({dispatch}) => {
  let input
  
  function onSubmit (event) {
    event.preventDefault()
    if (!input.value.trim()){
      return
    }

    dispatch(addTodo(input.value))
    input.value = ''
  }

  return (
    <div className='mb-4'>
      <form onSubmit={onSubmit}>
        <input className='shadow
          appearance-none
          border
          rounded
          py-2
          px-3
          md:w-2/4
          sm:w-full
          text-gray-700
          leading-tight
          focus:outline-none
          focus:shadow-outline
          m-4' ref={node => {input = node}}/>
        <button 
          className='font-bold
            py-2
            px-4
            rounded
            text-gray-900
            bg-gray-400
            hover:bg-gray-600
            md:w-1/4
            sm:w-full
            focus:bg-gray-500'
          type='submit'
          >
          <FontAwesomeIcon icon={faPlusCircle} className='mx-2' />
          Agregar tarea
        </button>
      </form>
    </div>
  )
}

export default connect()(AddTodo)
