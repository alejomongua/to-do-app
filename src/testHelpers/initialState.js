export default {
  alertas: [],
  todos: {
    list: ['0', '1', '2', '3', '4'],
    byId: {
      '0': {
        text: 'Tarea 1',
        completed: false,
      },
      '1': {
        text: 'Tarea 2',
        completed: true,
      },
      '2': {
        text: 'Tarea 3',
        completed: false,
      },
      '3': {
        text: 'Tarea 4',
        completed: true,
      },
      '4': {
        text: 'Tarea 5',
        completed: false,
      },
    }
  },
  visibilityFilter: 'SHOW_ALL',
}
