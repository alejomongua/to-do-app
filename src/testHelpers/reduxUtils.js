// test-utils.js
import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import _initialState from './initialState'
import store from '../redux/store'

const create = renderer.create

delete renderer.create

renderer.create = function (
  ui
) {
  return create(<Provider store={store}>{ui}</Provider>)
}

function render(
  ui
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper })
}

// re-export everything
export * from '@testing-library/react'
// override render method
export { render }
export { renderer }