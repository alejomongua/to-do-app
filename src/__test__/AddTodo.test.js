import '@babel/polyfill'
import React from 'react';
import AddTodo from '../components/AddTodo'
import { render, renderer, screen, fireEvent } from '../testHelpers/reduxUtils'
import _initialState from '../testHelpers/initialState'
import '@testing-library/jest-dom'

it('matches snapshot', () => {
  const tree = renderer
    .create(<AddTodo />)
    .toJSON();
  expect(tree).toMatchSnapshot()
})

it('has a button to add to do', () => {
  const { getByRole } = render(<AddTodo />)
  getByRole('button', { name: /agregar/i })
})

it('has an input to add to do', () => {
  const { getByRole } = render(<AddTodo />)
  getByRole('textbox')
})
