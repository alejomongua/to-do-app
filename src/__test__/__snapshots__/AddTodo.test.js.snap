// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="mb-4"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="shadow appearance-none border rounded py-2 px-3 md:w-2/4 sm:w-full text-gray-700 leading-tight focus:outline-none focus:shadow-outline m-4"
    />
    <button
      className="font-bold py-2 px-4 rounded text-gray-900 bg-gray-400 hover:bg-gray-600 md:w-1/4 sm:w-full focus:bg-gray-500"
      type="submit"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus-circle fa-w-16 mx-2"
        data-icon="plus-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      Agregar tarea
    </button>
  </form>
</div>
`;
